install/swiftly-install.sh

```diff
@@ -170,26 +167,198 @@ install_system_deps () {
+    PLATFORM_NAME="ubi$1"
+    PLATFORM_NAME_FULL="ubi$1"
+    docker_platform_name="rhel-ubi"
+    docker_platform_version="$1"
+    package_manager="yum"
+
+    if [[ -z "$PLATFORM_NAME_PRETTY" ]]; then
+        PLATFORM_NAME_PRETTY="RHEL 9"
+    fi
+}
+
+detect_platform () {
+    if [[ -f "/etc/os-release" ]]; then
+        OS_RELEASE="/etc/os-release"
+    elif [[ -f "/usr/lib/os-release" ]]; then
+        OS_RELEASE="/usr/lib/os-release"
+    else
+        manually_select_platform
+    fi
+
+    source "$OS_RELEASE"
+    PLATFORM_NAME_PRETTY="$PRETTY_NAME"
+
+    case "$ID$ID_LIKE" in
+        *"amzn"*)
+            if [[ "$VERSION_ID" != "2" ]]; then
+                manually_select_platform
+            else
+                set_platform_amazonlinux "2"
+            fi
+            ;;
+
+        *"ubuntu"*)
+            case "$UBUNTU_CODENAME" in
+                "jammy")
+                    set_platform_ubuntu "22" "04"
+                    ;;
+
+                "focal")
+                    set_platform_ubuntu "20" "04"
+                    ;;
+
+                "bionic")
+                    set_platform_ubuntu "18" "04"
+                    ;;
+
+                *)
+                    manually_select_platform
+                    ;;
+            esac
+            ;;
+
+        *"rhel"*)
+            if [[ "$VERSION_ID" != 9* ]]; then
+                manually_select_platform
+            else
+                set_platform_rhel "9"
+            fi
+            ;;
+
+        *)
+            manually_select_platform
+            ;;
+    esac
+}
+
+manually_select_platform () {
+    if [[ "$DISABLE_CONFIRMATION" == "true" ]]; then
+        echo "Error: Unsupported platform: $PRETTY_NAME"
+        exit 1
+    fi
+    echo "$PRETTY_NAME is not an officially supported platform, but the toolchains for another platform may still work on it."
+    echo ""
+    echo "Please select the platform to use for toolchain downloads:"
+
+    echo "0) Cancel"
+    echo "1) Ubuntu 22.04"
+    echo "2) Ubuntu 20.04"
+    echo "3) Ubuntu 18.04"
+    echo "4) RHEL 9"
+    echo "5) Amazon Linux 2"
+
+    read_input_with_default "0"
+    case "$READ_INPUT_RETURN" in
+        "1" | "1)")
+            set_platform_ubuntu "22" "04"
+            ;;
+
+        "2" | "2)")
+            set_platform_ubuntu "20" "04"
+            ;;
+
+        "3" | "3)")
+            set_platform_ubuntu "18" "04"
+            ;;
+
+        "4" | "4)")
+            set_platform_rhel "9"
+            ;;
+
+        "5" | "5)")
+            set_platform_amazonlinux "2"
+            ;;
+
+        *)
+            echo "Cancelling installation."
+            exit 0
+            ;;
+    esac
+}
+
+verify_getopt_install () {
+    if ! has_command "getopt" ; then
+        return 1
+    fi
+
+    getopt --test
+    # getopt --test exiting with status code 4 implies GNU getopt is being used, which we need.
+    [[ "$?" -eq 4 ]]
+    return "$?"
+}
+
+SWIFTLY_INSTALL_VERSION="0.2.0"
```

Comment:
the swiftly-install version doesn't need to be tied to swiftly's actual version imo, since we may make updates to it like this independent of changes to swiftly.

Reply:
We need to keep the tags on the repo in sync though. A new version of the script should be tagged a release, otherwise managing versions of the script is going to be painful.

Reply:
Yeah, I agree we ought to tag releases of swiftly-install, though we shouldn't mark them as a full on "Releases" in the GitHub sense of things, since then we'd need the install script to disambiguate when trying to download binaries.